src/core/reopt.jl

```diff
@@ -91,8 +145,19 @@ Solve the `Scenario` and `BAUScenario` in parallel using the first two (empty) m
 JSON file at the filepath `fp`.
 """
 function run_reopt(ms::AbstractArray{T, 1}, fp::String) where T <: JuMP.AbstractModel
-    d = JSON.parsefile(fp)
-    run_reopt(ms, d)
+
+	instantiate_logger()
+
+    try
+		d = JSON.parsefile(fp)
+    	run_reopt(ms, d)
```

Comment:
The code duplication is also leading to situations like this, where the logger is instantiated on line 149 and then again when run_reopt(ms, d) is called. Perhaps this case is harmless, but let's double check that there aren't more cases where reinstantiation of the logger makes messages get lost. For example, the one I pointed out in my comment on line 102.

Reply:
I have tested the logger instantiation function to make sure its working correctly now, so no duplication /overwriting of global logger. I have also moved the logger instantiation mainly to `Scenario` and `REoptInputs` since those always get called before `run_reopt`. I will update the testset to ensure that logger gets instantiated for most Julia runs and all API v3 runs.